packages/process/src/common/shell-command-builder.spec.ts

```diff
@@ -0,0 +1,450 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/**
+ * This test suite assumes that we run in a NodeJS environment!
+ */
+
+import { spawn, execSync, SpawnOptions, ChildProcess, spawnSync } from 'child_process';
+import { Readable } from 'stream';
+import { join } from 'path';
+
+import { ShellCommandBuilder, CommandLineOptions, ProcessInfo } from './shell-command-builder';
+
+import {
+    bgRed, bgWhite, bgYellow,
+    black, green, magenta, red, white, yellow,
+    bold,
+} from 'colors/safe'; // tslint:disable-line:no-implicit-dependencies
+
+export interface TestProcessInfo extends ProcessInfo {
+    shell: ChildProcess
+}
+
+const isWindows = process.platform === 'win32';
+const _debug = !!process.env['THEIA_TEST_DEBUG'];
```

Comment:
I don't like a new env variable which is used only by one test. Could we get rid of it and just always log necessary information but not more as we do for other tests?

Reply:
I really prefer being able to do extra logging in some way. The information being logged when this flag is enabled is useful when trying to understand what happens inside the shells, it is only really useful when developing the tests. Environment variables cost nothing, I don't plan on making this one standard, and anyone willing to investigate a test can quickly understand what he has to do with it: Set this var and you will get more info while the test is running.

Reply:
I usually rerun and debug only failing tests to see the reason, although agree that sometimes extra logging helps.